modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/AccountStatusInfo.kt

```diff
@@ -3,4 +3,5 @@ package au.com.shiftyjelly.pocketcasts.analytics
 interface AccountStatusInfo {
     fun isLoggedIn(): Boolean
     fun getUuid(): String?
+    fun signOut()
```

Comment:
This should be handled inside `SyncManagerImpl`. `AccountStatusInfo` is a simple data structure for analytics.

Reply:
I was thinking about this because none of them was fitting for this in my opinion. `SyncManager` reminds me synchronization things and `AccountStatusInfo` reminds me account things, but I also think it's not ideal because of the `*StatusInfo` in the end.

I can move it to `SyncManager`. It's not a big deal

Reply:
Looking at the code further it should probably be `SyncAccountManagerImpl` that handles it.

Reply:
When I was about to move the code, I noticed that `syncAccountManager.signOut()` was already performing this task but wasn't working due to the order in which the tasks inside the `action` block were being executed. To ensure that all tasks were executed sequentially, I wrapped them in a coroutine, and it worked as expected.

I did some regressions tests, but I would like to ask if you do the same if you can.

Reply:
I removed the unit tests from here since it was not related to this change anymore and added in another PR https://github.com/Automattic/pocket-casts-android/pull/2957

Reply:
Works as expected. Thanks!